client/RequestPacker.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <iostream>
+#include <vector>
+#include <stdexcept>
+#include <string>
+#include <cstring> // for memcpy
+
+// Base class for packing request packets
+class RequestPacketPacker {
+public:
+    // Constructor to initialize RequestPacketPacker fields
+    RequestPacketPacker(const std::vector<uint8_t>& clientId, uint8_t version, uint16_t code, const std::vector<uint8_t>& payload);
+

```

Comment:
base classes should declare a virtual desctructor. also should probably explicitly declare RO5